int16 material
int16 grip_mode
float32 coord
float32 size